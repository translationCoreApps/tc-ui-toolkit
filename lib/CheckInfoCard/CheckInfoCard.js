"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

require("./CheckInfoCard.styles.css");

var _PhraseWithToolTip = _interopRequireDefault(require("./PhraseWithToolTip"));

var CheckInfoCard = function CheckInfoCard(_ref) {
  var title = _ref.title,
      phrase = _ref.phrase,
      seeMoreLabel = _ref.seeMoreLabel,
      onSeeMoreClick = _ref.onSeeMoreClick,
      onLinkClick = _ref.onLinkClick,
      showSeeMoreButton = _ref.showSeeMoreButton,
      getScriptureFromReference = _ref.getScriptureFromReference;
  return _react["default"].createElement("div", {
    className: "checkInfo"
  }, _react["default"].createElement("div", {
    className: "leftSide"
  }, _react["default"].createElement("div", {
    className: "title"
  }, title)), _react["default"].createElement("div", {
    className: "rightSide"
  }, _react["default"].createElement("div", {
    className: "phrase"
  }, _react["default"].createElement(_PhraseWithToolTip["default"], {
    getScriptureFromReference: getScriptureFromReference,
    phrase: phrase,
    onClickLink: onLinkClick
  })), _react["default"].createElement("div", {
    onClick: showSeeMoreButton ? onSeeMoreClick : null,
    className: showSeeMoreButton ? 'linkActive' : 'linkInactive'
  }, seeMoreLabel)));
};

CheckInfoCard.propTypes = {
  phrase: _propTypes["default"].string,
  title: _propTypes["default"].string,
  seeMoreLabel: _propTypes["default"].string,
  onSeeMoreClick: _propTypes["default"].func,
  onLinkClick: _propTypes["default"].func,
  showSeeMoreButton: _propTypes["default"].bool,
  getScriptureFromReference: _propTypes["default"].func
};
var _default = CheckInfoCard;
exports["default"] = _default;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4uLy4uL3NyYy9DaGVja0luZm9DYXJkL0NoZWNrSW5mb0NhcmQuanMiXSwibmFtZXMiOlsiQ2hlY2tJbmZvQ2FyZCIsInRpdGxlIiwicGhyYXNlIiwic2VlTW9yZUxhYmVsIiwib25TZWVNb3JlQ2xpY2siLCJvbkxpbmtDbGljayIsInNob3dTZWVNb3JlQnV0dG9uIiwiZ2V0U2NyaXB0dXJlRnJvbVJlZmVyZW5jZSIsInByb3BUeXBlcyIsIlByb3BUeXBlcyIsInN0cmluZyIsImZ1bmMiLCJib29sIl0sIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7QUFBQTs7QUFDQTs7QUFFQTs7QUFDQTs7QUFHQSxJQUFNQSxhQUFhLEdBQUcsU0FBaEJBLGFBQWdCO0FBQUEsTUFDcEJDLEtBRG9CLFFBQ3BCQSxLQURvQjtBQUFBLE1BRXBCQyxNQUZvQixRQUVwQkEsTUFGb0I7QUFBQSxNQUdwQkMsWUFIb0IsUUFHcEJBLFlBSG9CO0FBQUEsTUFJcEJDLGNBSm9CLFFBSXBCQSxjQUpvQjtBQUFBLE1BS3BCQyxXQUxvQixRQUtwQkEsV0FMb0I7QUFBQSxNQU1wQkMsaUJBTm9CLFFBTXBCQSxpQkFOb0I7QUFBQSxNQU9wQkMseUJBUG9CLFFBT3BCQSx5QkFQb0I7QUFBQSxTQVNwQjtBQUFLLElBQUEsU0FBUyxFQUFDO0FBQWYsS0FDRTtBQUFLLElBQUEsU0FBUyxFQUFDO0FBQWYsS0FDRTtBQUFLLElBQUEsU0FBUyxFQUFDO0FBQWYsS0FDR04sS0FESCxDQURGLENBREYsRUFNRTtBQUFLLElBQUEsU0FBUyxFQUFDO0FBQWYsS0FDRTtBQUFLLElBQUEsU0FBUyxFQUFDO0FBQWYsS0FDRSxnQ0FBQyw2QkFBRDtBQUFtQixJQUFBLHlCQUF5QixFQUFFTSx5QkFBOUM7QUFBeUUsSUFBQSxNQUFNLEVBQUVMLE1BQWpGO0FBQXlGLElBQUEsV0FBVyxFQUFFRztBQUF0RyxJQURGLENBREYsRUFJRTtBQUFLLElBQUEsT0FBTyxFQUFFQyxpQkFBaUIsR0FBR0YsY0FBSCxHQUFvQixJQUFuRDtBQUF5RCxJQUFBLFNBQVMsRUFBRUUsaUJBQWlCLEdBQUcsWUFBSCxHQUFrQjtBQUF2RyxLQUNHSCxZQURILENBSkYsQ0FORixDQVRvQjtBQUFBLENBQXRCOztBQTBCQUgsYUFBYSxDQUFDUSxTQUFkLEdBQTBCO0FBQ3hCTixFQUFBQSxNQUFNLEVBQUVPLHNCQUFVQyxNQURNO0FBRXhCVCxFQUFBQSxLQUFLLEVBQUVRLHNCQUFVQyxNQUZPO0FBR3hCUCxFQUFBQSxZQUFZLEVBQUVNLHNCQUFVQyxNQUhBO0FBSXhCTixFQUFBQSxjQUFjLEVBQUVLLHNCQUFVRSxJQUpGO0FBS3hCTixFQUFBQSxXQUFXLEVBQUVJLHNCQUFVRSxJQUxDO0FBTXhCTCxFQUFBQSxpQkFBaUIsRUFBRUcsc0JBQVVHLElBTkw7QUFPeEJMLEVBQUFBLHlCQUF5QixFQUFFRSxzQkFBVUU7QUFQYixDQUExQjtlQVVlWCxhIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBQcm9wVHlwZXMgZnJvbSAncHJvcC10eXBlcyc7XG5cbmltcG9ydCAnLi9DaGVja0luZm9DYXJkLnN0eWxlcy5jc3MnO1xuaW1wb3J0IFBocmFzZVdpdGhUb29sVGlwIGZyb20gJy4vUGhyYXNlV2l0aFRvb2xUaXAnO1xuXG5cbmNvbnN0IENoZWNrSW5mb0NhcmQgPSAoe1xuICB0aXRsZSxcbiAgcGhyYXNlLFxuICBzZWVNb3JlTGFiZWwsXG4gIG9uU2VlTW9yZUNsaWNrLFxuICBvbkxpbmtDbGljayxcbiAgc2hvd1NlZU1vcmVCdXR0b24sXG4gIGdldFNjcmlwdHVyZUZyb21SZWZlcmVuY2UsXG59KSA9PiAoXG4gIDxkaXYgY2xhc3NOYW1lPVwiY2hlY2tJbmZvXCI+XG4gICAgPGRpdiBjbGFzc05hbWU9XCJsZWZ0U2lkZVwiPlxuICAgICAgPGRpdiBjbGFzc05hbWU9XCJ0aXRsZVwiPlxuICAgICAgICB7dGl0bGV9XG4gICAgICA8L2Rpdj5cbiAgICA8L2Rpdj5cbiAgICA8ZGl2IGNsYXNzTmFtZT1cInJpZ2h0U2lkZVwiPlxuICAgICAgPGRpdiBjbGFzc05hbWU9XCJwaHJhc2VcIj5cbiAgICAgICAgPFBocmFzZVdpdGhUb29sVGlwIGdldFNjcmlwdHVyZUZyb21SZWZlcmVuY2U9e2dldFNjcmlwdHVyZUZyb21SZWZlcmVuY2V9IHBocmFzZT17cGhyYXNlfSBvbkNsaWNrTGluaz17b25MaW5rQ2xpY2t9IC8+XG4gICAgICA8L2Rpdj5cbiAgICAgIDxkaXYgb25DbGljaz17c2hvd1NlZU1vcmVCdXR0b24gPyBvblNlZU1vcmVDbGljayA6IG51bGx9IGNsYXNzTmFtZT17c2hvd1NlZU1vcmVCdXR0b24gPyAnbGlua0FjdGl2ZScgOiAnbGlua0luYWN0aXZlJ30+XG4gICAgICAgIHtzZWVNb3JlTGFiZWx9XG4gICAgICA8L2Rpdj5cbiAgICA8L2Rpdj5cbiAgPC9kaXY+XG4pO1xuXG5DaGVja0luZm9DYXJkLnByb3BUeXBlcyA9IHtcbiAgcGhyYXNlOiBQcm9wVHlwZXMuc3RyaW5nLFxuICB0aXRsZTogUHJvcFR5cGVzLnN0cmluZyxcbiAgc2VlTW9yZUxhYmVsOiBQcm9wVHlwZXMuc3RyaW5nLFxuICBvblNlZU1vcmVDbGljazogUHJvcFR5cGVzLmZ1bmMsXG4gIG9uTGlua0NsaWNrOiBQcm9wVHlwZXMuZnVuYyxcbiAgc2hvd1NlZU1vcmVCdXR0b246IFByb3BUeXBlcy5ib29sLFxuICBnZXRTY3JpcHR1cmVGcm9tUmVmZXJlbmNlOiBQcm9wVHlwZXMuZnVuYyxcbn07XG5cbmV4cG9ydCBkZWZhdWx0IENoZWNrSW5mb0NhcmQ7XG4iXX0=