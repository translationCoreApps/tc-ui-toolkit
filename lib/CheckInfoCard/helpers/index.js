"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getOffset = getOffset;

function getOffset(el) {
  var _x = 0;
  var _y = 0;

  while (el && !isNaN(el.offsetLeft) && !isNaN(el.offsetTop)) {
    _x += el.offsetLeft - el.scrollLeft;
    _y += el.offsetTop - el.scrollTop;
    el = el.offsetParent;
  }

  return {
    top: _y,
    left: _x
  };
}
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4uLy4uLy4uL3NyYy9DaGVja0luZm9DYXJkL2hlbHBlcnMvaW5kZXguanMiXSwibmFtZXMiOlsiZ2V0T2Zmc2V0IiwiZWwiLCJfeCIsIl95IiwiaXNOYU4iLCJvZmZzZXRMZWZ0Iiwib2Zmc2V0VG9wIiwic2Nyb2xsTGVmdCIsInNjcm9sbFRvcCIsIm9mZnNldFBhcmVudCIsInRvcCIsImxlZnQiXSwibWFwcGluZ3MiOiI7Ozs7Ozs7QUFBTyxTQUFTQSxTQUFULENBQW1CQyxFQUFuQixFQUF1QjtBQUM1QixNQUFJQyxFQUFFLEdBQUcsQ0FBVDtBQUNBLE1BQUlDLEVBQUUsR0FBRyxDQUFUOztBQUVBLFNBQU9GLEVBQUUsSUFBSSxDQUFDRyxLQUFLLENBQUNILEVBQUUsQ0FBQ0ksVUFBSixDQUFaLElBQStCLENBQUNELEtBQUssQ0FBQ0gsRUFBRSxDQUFDSyxTQUFKLENBQTVDLEVBQTREO0FBQzFESixJQUFBQSxFQUFFLElBQUlELEVBQUUsQ0FBQ0ksVUFBSCxHQUFnQkosRUFBRSxDQUFDTSxVQUF6QjtBQUNBSixJQUFBQSxFQUFFLElBQUlGLEVBQUUsQ0FBQ0ssU0FBSCxHQUFlTCxFQUFFLENBQUNPLFNBQXhCO0FBQ0FQLElBQUFBLEVBQUUsR0FBR0EsRUFBRSxDQUFDUSxZQUFSO0FBQ0Q7O0FBQ0QsU0FBTztBQUFFQyxJQUFBQSxHQUFHLEVBQUVQLEVBQVA7QUFBV1EsSUFBQUEsSUFBSSxFQUFFVDtBQUFqQixHQUFQO0FBQ0QiLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgZnVuY3Rpb24gZ2V0T2Zmc2V0KGVsKSB7XG4gIHZhciBfeCA9IDA7XG4gIHZhciBfeSA9IDA7XG5cbiAgd2hpbGUgKGVsICYmICFpc05hTihlbC5vZmZzZXRMZWZ0KSAmJiAhaXNOYU4oZWwub2Zmc2V0VG9wKSkge1xuICAgIF94ICs9IGVsLm9mZnNldExlZnQgLSBlbC5zY3JvbGxMZWZ0O1xuICAgIF95ICs9IGVsLm9mZnNldFRvcCAtIGVsLnNjcm9sbFRvcDtcbiAgICBlbCA9IGVsLm9mZnNldFBhcmVudDtcbiAgfVxuICByZXR1cm4geyB0b3A6IF95LCBsZWZ0OiBfeCB9O1xufSJdfQ==