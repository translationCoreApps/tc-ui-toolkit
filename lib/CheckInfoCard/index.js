"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _CheckInfoCard["default"];
  }
});

var _CheckInfoCard = _interopRequireDefault(require("./CheckInfoCard"));
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4uLy4uL3NyYy9DaGVja0luZm9DYXJkL2luZGV4LmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7Ozs7Ozs7Ozs7Ozs7O0FBQUEiLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgeyBkZWZhdWx0IH0gZnJvbSAnLi9DaGVja0luZm9DYXJkJztcbiJdfQ==