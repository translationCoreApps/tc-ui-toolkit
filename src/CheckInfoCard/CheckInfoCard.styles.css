@import '../index.styles.css';

.handleIcon {
  z-index: 999;
  color: var(--reverse-color);
  background-color: var(--text-color-dark);
  padding: 10px 0px;
  border-radius: 5px 0px 0px 5px
}

.linkActive {
  font-weight: bold;
  display: flex;
  justify-content: flex-end;
  margin: 5px 20px;
  cursor: pointer
}

.linkInactive {
  font-weight: bold;
  color: var(--text-color);
  margin: 5px 20px;
  text-align: right;
  cursor: default;
}

.checkInfo {
  flex: 0 0 100px;
  display: flex;
  margin: 0 10px;
  color: var(--reverse-color);
  background-color: var(--accent-color-dark);
  box-shadow: 0 3px 10px var(--background-color);
  border-radius: 2px;
}

.leftSide {
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
}

.rightSide {
  flex: 2;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-top: 10px;
  border-left: 1px solid var(--reverse-color);
}

.title {
  max-height: 100px;
  overflow-y: auto;
  padding: 0 20px;
  font-size: 16px;
  font-weight: bold;
  text-align: center;
}

.phrase {
  max-height: 80px;
  overflow-y: auto;
  padding: 0 20px;
  text-align: center;
}
