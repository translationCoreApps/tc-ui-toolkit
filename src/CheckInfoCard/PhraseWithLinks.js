import React from 'react';
import ReactTooltip from "react-tooltip";
import PropTypes from "prop-types";

const LinkableThelpsResources = ['ta', 'tw'];

const PhraseWithLinks = ({phrase, getScriptureFromReference, onTHelpsLinkClick}) => {
  // The split regular expression. NOTE: the outside parans are what keep the match in the splitPhrase array
  // This matches a Markdown link in the form of (Title)[rc://lang/resource/type/ending]
  const splitRE = /(\[[^)]+]\s*\(rc:\/\/[\w-]+\/[\w-]+\/[\w-]+\/[^\s)\]]+\))/;
  // splitPhrase will be an array of alternating [text, link, text, link, etc.], thus even index are text, odd are links
  const splitPhrase = phrase.split(splitRE);
  if (splitPhrase.length < 2)
    return phrase;
  // Now go through those text elements and link elements and put them into a an array of components
  const textAndLinks = splitPhrase.reduce((prev, current, i) => {
    // the first element starts the new list and is text.
    const key = 'phrase-' + i;
    if (i === 0)
      return [(<span key={key}>{current}</span>)];
    // All even elements are text, so leave as is.
    if (!(i % 2))
      return prev.concat(<span key={key}>{current}</span>);
    // Now we have an odd indexed element, which is a link. If its a Bible reference, we make it a tooltip.
    // if it is a tHelps link, we make it clickable to open in the tHelps sidebar.
    // Any other link we will have it be a tooltip as the rc link.
    // Again, this is matching (Title)[rc://lang/resource/type/ending] but getting each part
    const linkRE = /\[([^)]+)]\s*\((rc:\/\/([\w-]+)\/([\w-]+)\/([\w-]+)\/([^)]+))\)/;
    let match = current.match(linkRE);
    if (!match)
      // Shouldn't ever be here, but just in case
      return prev.concat(<span key={key}>{current}</span>);
    match.shift();
    const [title, wholeLink, lang, resource, type, ending] = match;
    if (type === 'book') {
      // It's a Bible reference
      const bibleRefRE = /(\w+)\/(\d+)\/(\d+)/;
      const match = ending.match(bibleRefRE);
      if (match) {
        // It properly parsed from <book>/<chapter>/<verse>
        match.shift();
        const [book, chapter, verse] = match;
        const tooltip = getScriptureFromReference(lang, resource, book, chapter, verse);
        return prev.concat(<span
          key={key}
          className="scripture"
          data-for="phrase-tooltip"
          data-tip={tooltip}
          data-place="bottom"
          data-effect="float"
          data-type="light"
          style={{minWidth: 0}}
          data-class="phrase-scripture-tooltip"
          data-delay-hide="100">
            <span className={'phrase-scripture-title'} style={{whiteSpace: 'nowrap', textDecoration: 'underline'}}>
              {title}
            </span>
        </span>);
      }
    }
    if (LinkableThelpsResources.indexOf(resource) >= 0) {
      // It's a tHelps link, so we call the function to show the article in the tHelps modal
      const link = [lang, resource, ending].join('/');
      return prev.concat(
        <span
          key={key}
          className={'thelps-link-title'}
          style={{whiteSpace: 'nowrap', textDecoration: 'underline'}}
          onClick={() => onTHelpsLinkClick(link)}
        >
          {title}
        </span>
      );
    }
    // Leaves the link as tooltip to the title
    return prev.concat(<span
      key={key}
      className="scripture"
      data-for="phrase-tooltip"
      data-tip={wholeLink}
      data-place="bottom"
      data-effect="float"
      data-type="light"
      style={{minWidth: 0}}
      data-class="phrase-link-tooltip"
      data-delay-hide="100">
            <span className={'phrase-link-title'} style={{whiteSpace: 'nowrap', textDecoration: 'underline'}}>
              {title}
            </span>
      </span>);
  }, []);
  return (<div>
    {textAndLinks}
    <ReactTooltip id="phrase-tooltip"/>
  </div>);
};

PhraseWithLinks.propTypes = {
  phrase: PropTypes.string.isRequired,
  getScriptureFromReference: PropTypes.func.isRequired,
  onTHelpsLinkClick: PropTypes.func.isRequired,
};

export default PhraseWithLinks;
