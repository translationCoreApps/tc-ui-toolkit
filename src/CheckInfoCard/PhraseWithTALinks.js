import React, { Component } from 'react';
import PropTypes from 'prop-types';

class PhraseWithTALinks extends Component {
  constructor(props) {
    super(props);
  }

  onTALinkClick = (link) => {
    window.followLink = link;
  };

  render() {
    const {phrase} = this.props;
    if (!phrase)
      return phrase;
    const regex = /\[\[rc:\/\/([^/]+\/ta\/man\/[^\]]+)]]/;
    const splitPhrase = phrase.split(regex);
    if (splitPhrase.length < 2)
      return phrase;
    console.log("split", splitPhrase);
    const textAndLinks = splitPhrase
      .reduce((prev, current, i) => {
        if (!i)
          return [current];
        if (!(i % 2)) {
          return prev.concat(current);
        }
        return prev.concat(
          <a onClick={()=>this.onTALinkClick(current)}>
            {current}
          </a>
        );
      }, []);
    console.log(textAndLinks);
    return (<div>
      {textAndLinks}
    </div>);
  }
}

PhraseWithTALinks.propTypes = {
  phrase: PropTypes.string.isRequired,
};

export default PhraseWithTALinks;
