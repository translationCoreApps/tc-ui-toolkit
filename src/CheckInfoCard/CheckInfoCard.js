import React from 'react';
import PropTypes from 'prop-types';

import './CheckInfoCard.styles.css';
import {getOffset} from './helpers';


const CheckInfoCard = ({
  title,
  phrase,
  seeMoreLabel,
  onSeeMoreClick,
  showSeeMoreButton,
  getScriptureFromReference
}) => {
  /** @type HTMLElement */
  let scriptureRef;
  let popOverRef;
  let phraseWithPopover;
  const rcMatch = phrase.match(/\[(\w+.+:\d+)]\(rc:\/\/(\w+)\/(\w+)\/book\/(\w+)\/(\d+)\/(\d+)\)/) || [];
  const showPopover = rcMatch.length > 0;
  if (showPopover) {
    const [wholeMatch, referenceText , lang, id, book, chapter, verse] = rcMatch;
    const [preReference, postReference] = phrase.split(wholeMatch);
    const popoverLabel = getScriptureFromReference(lang, id, book, chapter, verse);
    phraseWithPopover = (<div>
      {preReference}
      <span onMouseEnter={() => {
        const {top, left} = getOffset(scriptureRef);
        popOverRef.style.top = `${top}px`;
        popOverRef.style.left = `${left}px`;
        popOverRef.style.width = `${scriptureRef.offsetWidth}px`;
        popOverRef.style.height = `${scriptureRef.offsetHeight}px`;
      }} style={{position: 'relative'}}>
        <div ref={(ref) => popOverRef = ref} aria-label={popoverLabel} className="hint--top hint--medium" style={{
          position: 'fixed'
        }} />
        <span style={{
          whiteSpace: 'nowrap',
          textDecoration: 'underline'
        }} ref={(ref) => scriptureRef = ref}>{referenceText}</span>
      </span>
      {postReference}
    </div>
    );
  }
  return (
    <div className="checkInfo">
      <div className="leftSide">
        <div className="title">
          {title}
        </div>
      </div>
      <div className="rightSide">
        <div className="phrase">
          {phraseWithPopover || phrase}
        </div>
        <div onClick={showSeeMoreButton ? onSeeMoreClick : null} className={showSeeMoreButton ? 'linkActive' : 'linkInactive'}>
          {seeMoreLabel}
        </div>
      </div>
    </div>
  );
};

CheckInfoCard.propTypes = {
  phrase: PropTypes.string,
  title: PropTypes.string,
  seeMoreLabel: PropTypes.string,
  onSeeMoreClick: PropTypes.func,
  showSeeMoreButton: PropTypes.bool,
  getScriptureFromReference: PropTypes.func
};

export default CheckInfoCard;
