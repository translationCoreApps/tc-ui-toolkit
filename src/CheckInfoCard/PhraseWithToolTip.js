import React from 'react';
import {getOffset} from './helpers';

function PhraseWithToolTip({phrase, getScriptureFromReference}) {
  let scriptureRef;
  let tooltipRef;
  const rcMatch = phrase.match(/\[(\w+.+:\d+)]\(rc:\/\/(\w+)\/(\w+)\/book\/(\w+)\/(\d+)\/(\d+)\)/) || [];
  const showTooltip = rcMatch.length > 0;
  if (showTooltip) {
    const [wholeMatch, referenceText, lang, id, book, chapter, verse] = rcMatch;
    const [preReference, postReference] = phrase.split(wholeMatch);
    const tooltipLabel = getScriptureFromReference(lang, id, book, chapter, verse);
    return (<div>
      {preReference}
      <span onMouseEnter={() => {
        const {top, left} = getOffset(scriptureRef);
        tooltipRef.style.top = `${top}px`;
        tooltipRef.style.left = `${left}px`;
        tooltipRef.style.width = `${scriptureRef.offsetWidth}px`;
        tooltipRef.style.height = `${scriptureRef.offsetHeight}px`;
      }} style={{position: 'relative'}}>
        <div ref={(ref) => tooltipRef = ref} aria-label={tooltipLabel} className="hint--top hint--medium" style={{
          position: 'fixed'
        }} />
        <span style={{
          whiteSpace: 'nowrap',
          textDecoration: 'underline'
        }} ref={(ref) => scriptureRef = ref}>{referenceText}</span>
      </span>
      {postReference}
    </div>
    );
  } else {
    return phrase;
  }
}

export default PhraseWithToolTip;
