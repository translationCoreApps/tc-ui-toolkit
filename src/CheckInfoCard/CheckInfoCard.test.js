import React from 'react';
import renderer from 'react-test-renderer';
import CheckInfoCard from './CheckInfoCard';


describe('CheckInfoCard', () => {
  it('renders correctly', () => {
    const props = {
      title: 'save, saves, saved, safe, salvation',
      phrase: 'The term "save" refers to keeping someone from experiencing something bad or harmful. To "be safe" means to be protected from harm or danger.',
      seeMoreLabel: 'See More',
      showSeeMoreButton: true,
      onSeeMoreClick: () => {},
    };
    const tree = renderer
      .create(
        <CheckInfoCard
          title={props.title}
          phrase={props.phrase}
          seeMoreLabel={props.seeMoreLabel}
          showSeeMoreButton={props.showSeeMoreButton}
          onSeeMoreClick={props.onSeeMoreClick}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
