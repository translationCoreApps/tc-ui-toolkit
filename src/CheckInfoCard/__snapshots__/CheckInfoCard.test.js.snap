// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckInfoCard renders correctly 1`] = `
<div
  className="checkInfo"
>
  <div
    className="leftSide"
  >
    <div
      className="title"
    >
      save, saves, saved, safe, salvation
    </div>
  </div>
  <div
    className="rightSide"
  >
    <div
      className="phrase"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>The term &quot;save&quot; refers to keeping someone from experiencing something bad or harmful. To &quot;be safe&quot; means to be protected from harm or danger.</p>
",
          }
        }
        style={
          Object {
            "color": "#fff",
          }
        }
      />
    </div>
    <div
      className="linkActive"
      onClick={[Function]}
    >
      See More
    </div>
  </div>
</div>
`;
